### ISSUE TITLE
Kong ingress controller restart multiple times

### ISSUE THREAD

**Issue by xpunch** [Kong/kubernetes-ingress-controller#1346]
### Summary

SUMMARY_GOES_HERE

**Kong Ingress controller version**
kong:2.3
kong/kubernetes-ingress-controller:1.2

### Kubernetes version
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:54Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

### Environment
EKS 18.9

### What happened
Kong restart multiple times
```
Name:                 ingress-kong-694cdf9ff6-j5d72
Namespace:            kong
Priority Class Name:  system-node-critical
Node:                 fargate-ip
Start Time:           Fri, 21 May 2021 15:54:30 +0800
Labels:               app=ingress-kong
Annotations:      kubernetes.io/psp: eks.privileged
                      kuma.io/gateway: enabled
                      prometheus.io/port: 8100
                      prometheus.io/scrape: true
                      traffic.sidecar.istio.io/includeInboundPorts: 
Status:               Running
Controlled By:        ReplicaSet/ingress-kong
Containers:
  proxy:
    Container ID:   containerd://0b439f7c6c4edc30778c4fb3304253c9e013762019ff8a46f357ba718dcd9f53
    Image:          kong:2.3
    Image ID:       docker.io/library/kong@sha256:d4c036d82d35e0c56a5a7dc1d018a92e97b7a9540cb202dd39753e70bac72de9
    Ports:          8000/TCP, 8443/TCP, 8100/TCP
    Host Ports:     0/TCP, 0/TCP, 0/TCP
    State:          Running
      Started:      Tue, 25 May 2021 15:27:57 +0800
    Last State:     Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 25 May 2021 06:20:53 +0800
      Finished:     Tue, 25 May 2021 15:27:56 +0800
    Ready:          True
    Restart Count:  8
    Liveness:   http-get http://:8100/status delay=5s timeout=1s period=10s #success=1 #failure=3
    Readiness:  http-get http://:8100/status delay=5s timeout=1s period=10s #success=1 #failure=3
    Environment:
      KONG_PROXY_LISTEN:             0.0.0.0:8000 proxy_protocol, 0.0.0.0:8443 ssl proxy_protocol
      KONG_TRUSTED_IPS:              0.0.0.0/0,::/0
      KONG_REAL_IP_HEADER:           proxy_protocol
      KONG_PORT_MAPS:                80:8000, 443:8443
      KONG_ADMIN_LISTEN:             127.0.0.1:8444 ssl
      KONG_STATUS_LISTEN:            0.0.0.0:8100
      KONG_DATABASE:                 off
      KONG_NGINX_WORKER_PROCESSES:   2
      KONG_ADMIN_ACCESS_LOG:         /dev/stdout
      KONG_ADMIN_ERROR_LOG:          /dev/stderr
      KONG_PROXY_ERROR_LOG:          /dev/stderr
  ingress-controller:
    Container ID:   containerd://e8ae48e9a93b792e8c7fa11d10bf9e47da8085e520221af130a324bca0544d7f
    Image:          kong/kubernetes-ingress-controller:1.2
    Image ID:       docker.io/kong/kubernetes-ingress
    Port:           8080/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Tue, 25 May 2021 15:27:58 +0800
    Last State:     Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 25 May 2021 06:20:54 +0800
      Finished:     Tue, 25 May 2021 15:27:57 +0800
    Ready:          True
    Restart Count:  8
    Liveness:   http-get http://:10254/healthz delay=5s timeout=1s period=10s #success=1 #failure=3
    Readiness:  http-get http://:10254/healthz delay=5s timeout=1s period=10s #success=1 #failure=3
    Environment:
      CONTROLLER_KONG_ADMIN_URL:              https://127.0.0.1:8444
      CONTROLLER_KONG_ADMIN_TLS_SKIP_VERIFY:  true
      CONTROLLER_PUBLISH_SERVICE:             kong/kong-proxy
      KONG_REAL_IP_HEADER:                    X-Forwarded-For
      KONG_REAL_IP_RECURSIVE:                 on
      POD_NAME:                               ingress-kong-694cdf9ff6-j5d72 (v1:metadata.name)
      POD_NAMESPACE:                          kong (v1:metadata.namespace)
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kong-serviceaccount-token-7jqn4 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:          <none>
```

### Expected behavior
Kong works fine

### Logs before kong restarted
```
E0525 07:27:12.620178       1 leaderelection.go:361] Failed to update lock: context deadline exceeded
I0525 07:27:13.725337       1 leaderelection.go:278] failed to renew lease kong/ingress-controller-leader-kong: timed out waiting for the condition
time="2021-05-25T07:27:15Z" level=info msg="stopped leading" component=status-syncer
time="2021-05-25T07:27:16Z" level=warning msg="leader election session terminated unexpectedly" component=elector delay=1s retry-count=2
time="2021-05-25T07:27:15Z" level=error msg="polling failed :timed out waiting for the condition" component=status-syncer
time="2021-05-25T07:27:18Z" level=info msg="starting leader election" component=elector retry-count=2
I0525 07:27:18.819209       1 leaderelection.go:243] attempting to acquire leader lease kong/ingress-controller-leader-kong...
I0525 07:27:25.814263       1 leaderelection.go:253] successfully acquired lease kong/ingress-controller-leader-kong
time="2021-05-25T07:27:26Z" level=info msg="started leading" component=status-syncer
time="2021-05-25T07:27:30Z" level=info msg="syncing configuration" component=controller
E0525 07:27:56.121986       1 leaderelection.go:361] Failed to update lock: context deadline exceeded
I0525 07:27:56.122759       1 leaderelection.go:278] failed to renew lease kong/ingress-controller-leader-kong: timed out waiting for the condition
time="2021-05-25T07:27:56Z" level=info msg="no configuration change, skipping sync to kong" component=controller
time="2021-05-25T07:27:56Z" level=info msg="stopped leading" component=status-syncer
time="2021-05-25T07:27:56Z" level=warning msg="leader election session terminated unexpectedly" component=elector delay=1s retry-count=3
time="2021-05-25T07:27:56Z" level=error msg="polling failed :timed out waiting for the condition" component=status-syncer
time="2021-05-25T07:27:56Z" level=info msg="Received SIGTERM, shutting down" component=signal-handler
time="2021-05-25T07:27:56Z" level=info msg="shutting down controller queues" component=controller
time="2021-05-25T07:27:56Z" level=info msg="awaiting completion of shutdown procedures" component=controller
```

**Comment by xpunch** [comment]
Suddenly return 404 for every request.
no Route matched with those values
Works fine for 17d.

**Comment by xpunch** [comment]
Problem fixed after upgrade to kong:2.4
